src/zapv2/__init__.py

```diff
@@ -146,7 +147,18 @@ def _request_api(self, url, query=None):
           # Add the apikey to get params for backwards compatibility
           if not query.get('apikey'):
             query['apikey'] = self.__apikey
-        return self.session.get(url, params=query, proxies=self.__proxies, verify=False)
+
+        response = self.session.get(url, params=query, proxies=self.__proxies, verify=False)
+
+        if (self.__validate_status_code and response.status_code >= 300 and response.status_code < 500):
+            raise Exception("Invalid status code returned from zap, which indicate a bad request: " 
```

Comment:
- `Error` instead of `Invalid`?
 - `indicates`?
 - "ZAP" should be all upper case.

(Same comments for the other message.)

Reply:
What do you think about `Non-successfull status code`?

Reply:
Looks better :)